coverage_point_calculator/src/lib.rs

```diff
@@ -0,0 +1,933 @@
+                    signal_level: SignalLevel::High,
+                    assignments: assignments_maximum(),
+                    boosted: None,
+                },
+                RankedCoverage {
+                    hotspot_key: pubkey(),
+                    cbsd_id: None,
+                    hex: hex_location(),
+                    rank: 42,
+                    signal_level: SignalLevel::High,
+                    assignments: assignments_maximum(),
+                    boosted: None,
+                },
+            ],
+        )
+        .expect("indoor wifi");
+
+        assert_eq!(
+            dec!(400),
+            calculate_coverage_points(indoor_wifi).total_coverage_points
+        );
+    }
+
+    #[test]
+    fn location_trust_score_multiplier() {
+        // Location scores are averaged together
+        let indoor_wifi = RewardableRadio::new(
+            RadioType::IndoorWifi,
+            speedtest_maximum(),
+            location_trust_with_scores(&[dec!(0.1), dec!(0.2), dec!(0.3), dec!(0.4)]),
+            RadioThreshold::Verified,
+            vec![RankedCoverage {
+                hotspot_key: pubkey(),
+                cbsd_id: None,
+                hex: hex_location(),
+                rank: 1,
+                signal_level: SignalLevel::High,
+                assignments: assignments_maximum(),
+                boosted: None,
+            }],
+        )
+        .expect("indoor wifi");
+
+        // Location trust scores is 1/4
+        // (0.1 + 0.2 + 0.3 + 0.4) / 4
+        assert_eq!(
+            dec!(100),
+            calculate_coverage_points(indoor_wifi).total_coverage_points
+        );
+    }
+
+    #[test]
+    fn boosted_hex() {
+        let covered_hexes = vec![
+            RankedCoverage {
+                hotspot_key: pubkey(),
+                cbsd_id: None,
+                hex: hex_location(),
+                rank: 1,
+                signal_level: SignalLevel::High,
+                assignments: assignments_maximum(),
+                boosted: None,
+            },
+            RankedCoverage {
+                hotspot_key: pubkey(),
+                cbsd_id: None,
+                hex: hex_location(),
+                rank: 1,
+                signal_level: SignalLevel::Low,
+                assignments: assignments_maximum(),
+                boosted: NonZeroU32::new(4),
+            },
+        ];
+        let indoor_wifi = RewardableRadio::new(
+            RadioType::IndoorWifi,
+            speedtest_maximum(),
+            location_trust_maximum(),
+            RadioThreshold::Verified,
+            covered_hexes.clone(),
+        )
+        .expect("indoor wifi");
+
+        // The hex with a low signal_level is boosted to the same level as a
+        // signal_level of High.
+        assert_eq!(
+            dec!(800),
+            calculate_coverage_points(indoor_wifi.clone()).total_coverage_points
+        );
+    }
+
+    #[test]
+    fn base_radio_coverage_points() {
```

Comment:
This feels like it should be 4 individual tests

Reply:
I broke this test apart, and parameterized it by SignalLevel.
If any coverage points change for hexes or radio type, a unique test will break. :+1: